### PR TITLE
Fix pingbacks in Postgres

### PR DESCRIPTION
**Pull request by oetherington**
This PR fixes pingbacks on the EA Forum which were broken when we migrated posts to Postgres. The solution here is to add a new mongo-like operator that is only used when we're in Postgres which can correctly generate the SQL without requiring schemas for deep JSON objects. This is a bit of a hack, but it's also pretty rare in the code base.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203859716354875) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/utils/viewUtils.ts [hunk 1]
```diff
@@ -157,3 +157,10 @@ function replaceSpecialFieldSelectors(selector: any): any {
   return result;
 }
 
+export const jsonArrayContainsSelector = <T extends DbObject>(
+  collection: CollectionBase<T>,
+  field: string,
+  value: any,
+) => collection.isPostgres()
+  ? {$expr: {$jsonArrayContains: [field, value]}}
+  : {[field]: value};
```
Comment (jpaddison3) [inline packages/lesswrong/lib/utils/viewUtils.ts:166]: This is cute. It's hacky, but it's cute.

**Comment by jpaddison3** [comment]
> also pretty rare in the code base

Just to check — this is the only instance? Or maybe: "It's definitely the only instance I know of, but I wouldn't put it past the codebase to contain another".

**Review by jpaddison3 — APPROVED** [review]
LGTM 🙏

**Comment by oetherington** [comment]
> > also pretty rare in the code base
> 
> Just to check — this is the only instance? Or maybe: "It's definitely the only instance I know of, but I wouldn't put it past the codebase to contain another".

100% the latter - I'm pretty sure it's ungrepable

**Comment by jpaddison3** [comment]
👍 Does seem entirely ungrepable

**Pull request merged**